integration/k8s_test.go

```diff
@@ -0,0 +1,118 @@
+package integration
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/containous/traefik/integration/try"
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/testhelpers"
+	"github.com/go-check/check"
+	checker "github.com/vdemeester/shakers"
+	v1 "k8s.io/api/core/v1"
+	v1beta12 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// K8sSuite
+type K8sSuite struct{ BaseSuite }
+
+func (s *K8sSuite) SetUpSuite(c *check.C) {
+	s.createComposeProject(c, "k8s")
+	s.composeProject.Start(c)
+}
+
+func (s *K8sSuite) TearDownSuite(c *check.C) {
+	s.composeProject.Stop(c)
+	os.Remove("./resources/compose/output/kubeconfig.yaml")
+}
+
+func parseK8sYaml(fileR []byte) []runtime.Object {
+	acceptedK8sTypes := regexp.MustCompile(`(Deployment|Service|Ingress)`)
+	sepYamlfiles := strings.Split(string(fileR), "---")
+	retVal := make([]runtime.Object, 0, len(sepYamlfiles))
+	for _, f := range sepYamlfiles {
+		if f == "\n" || f == "" {
+			continue
+		}
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		obj, groupVersionKind, err := decode([]byte(f), nil, nil)
+
+		if err != nil {
+			log.WithoutContext().Debugf("Error while decoding YAML object. Err was: %s", err)
+			continue
+		}
+
+		if !acceptedK8sTypes.MatchString(groupVersionKind.Kind) {
+			log.WithoutContext().Debugf("The custom-roles configMap contained K8s object types which are not supported! Skipping object with type: %s", groupVersionKind.Kind)
+		} else {
+			retVal = append(retVal, obj)
+		}
+	}
+	return retVal
+}
+
+func (s *K8sSuite) TestSimpleDefaultConfig(c *check.C) {
+	req := testhelpers.MustNewRequest(http.MethodGet, "https://127.0.0.1:6443", nil)
+	err := try.RequestWithTransport(req, time.Second*60, &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}, try.StatusCodeIs(http.StatusUnauthorized))
+	c.Assert(err, checker.IsNil)
+
+	abs, err := filepath.Abs("./resources/compose/output/kubeconfig.yaml")
+	c.Assert(err, checker.IsNil)
+
+	err = try.Do(time.Second*60, try.DoCondition(func() error {
```

Comment:
should `time.Second*60` be hard coded?

Reply:
I think we nearly never used constant for time in integration tests.
so :thinking: